clang/runtime/dpct-rt/include/dpct/device.hpp

```diff
@@ -659,19 +658,51 @@ class dev_mgr {
   void select_device(unsigned int id) {
     std::lock_guard<std::recursive_mutex> lock(m_mutex);
     check_id(id);
-    _thread2dev_map[get_tid()]=id;
+    _thread2dev_map[get_tid()] = id;
   }
   unsigned int device_count() { return _devs.size(); }
 
   unsigned int get_device_id(const sycl::device &dev) {
     unsigned int id = 0;
-    for(auto dev_item : _devs) {
+    for (auto dev_item : _devs) {
       if (*dev_item == dev) {
-        break;
+        return id;
       }
       id++;
     }
-    return id;
+    throw std::runtime_error(
+        "The device[" + dev.get_info<sycl::info::device::name>() +
+        "] is filtered out by dev_mgr::filter/filter_device in current device "
+        "list!");
+  }
+
+  /// Filter out devices; only keep the device with the name specified.
+  /// May break device id mapping and change current device.
```

Comment:
For L680, please be more specific on what is: id mapping.  suggest provide a func to print the dev-id mapping.